src/t8_forest/t8_forest_general.h

```diff
@@ -363,10 +365,22 @@ t8_forest_set_ghost (t8_forest_t forest, int do_ghost, t8_ghost_type_t ghost_typ
  *                                If 2, the iterative algorithm for unbalanced forests.
  *                                If 3, the top-down search algorithm for unbalanced forests.
  * \see t8_forest_set_ghost
+ * \note this function creates an ghost_interface obcejct and call \ref t8_forest_set_ghost_ext_new
  */
 void
 t8_forest_set_ghost_ext (t8_forest_t forest, int do_ghost, t8_ghost_type_t ghost_type, int ghost_version);
 
+/** Set a ghost_interface
+ * In application schoud only used if the user creates its own ghost_interface class (type = userderdefined)
+ * \param [in]    forest          the fores
+ * \param [in]    do_ghost        ---
+ * \param [in]    ghost_interface pointer to an object of the class ghost_interface or a derived class
+ *                                The forest takes ownership of the ghost_interface
+ * \note if the forest has already a ghost_interface, the old one will be unref and the new one will be set.
+*/
+void
+t8_forest_set_ghost_ext_new (t8_forest_t forest, int do_ghost, t8_forest_ghost_interface_c *ghost_interface);
```

Comment:
```suggestion
t8_forest_set_ghost_ext_new (t8_forest_t forest, const int do_ghost, const t8_forest_ghost_interface_c *ghost_interface);
```

Reply:
In this function we set `forest->ghost_interface = ghost_interface;` and if I make the variable `ghost_interface` of the function const, I can't write it to the forest, because it expects no const member.